zookeeper/core/component.py

```diff
@@ -277,6 +277,64 @@ def wrapped_fn(instance) -> List[str]:
     component_cls.__dir__ = wrapped_fn
 
 
+def _wrap_configure(component_cls: Type) -> None:
+    if hasattr(component_cls, "__configure__"):
+        if not callable(component_cls.__configure__):
+            raise TypeError(
+                "The `__configure__` attribute of a @component class must be a "
+                "method."
+            )
+        call_args = inspect.signature(component_cls.__configure__).parameters
+        configure_args = inspect.signature(configure).parameters
+        configure_args = {
+            "self": configure_args["instance"],
+            **{k: v for k, v in configure_args.items() if not k == "instance"},
+        }
+
+        error_message = (
+            "The `__configure__` method of a @component class must match the arguments "
+            f"of `configure()`, but `{component_cls.__name__}.__configure__` "
+            f"accepts arguments {tuple(name for name in call_args)}. Valid arguments: "
+            f"({', '.join(name for name in configure_args)}, *args, **kwargs)"
+        )
+
+        for arg in call_args:
+            if arg in ["self", "args", "kwargs"]:
+                continue
+            if arg not in configure_args:
+                raise TypeError(error_message)
+
+        for arg in configure_args:
+            if (
+                arg not in call_args
+                and "args" not in call_args
+                and "kwargs" not in call_args
+            ):
+                raise TypeError(error_message)
```

Comment:
This error handling logic would incorrectly throw for the following example, because it relies on the names of the catch-all arguments:
```python
def __configure__(self, *some_args, **some_kwargs):
    configure(self, *some_args, **some_kwargs)
```

I think a solution would be:
```suggestion
        for arg_name, arg_param in configure_args.items():
            if (
                arg_param.kind in (arg_param.VAR_POSITIONAL, arg_param.VAR_KEYWORD) or
                arg_name in configure_args
            ):
                continue
            raise TypeError(error_message)
```

This admittedly doesn't test quite as many cases. In particular, it only raises an error if an argument is incorrectly included, not if one is incorrectly omitted. But I think fewer false-positive error messages is a good thing.

Reply:
This would not work, since all of them are of kind `VAR_POSITIONAL_OR_KEYWORD`...